docs/guides/service_configuration_file.md

```diff
@@ -0,0 +1,304 @@
+type: skill
+<agent_number>:
+  <overridden_attributes>
+```
+
+For example, if we wish that each agent in the [Hello World service](../demos/hello_world_demo.md) outputs a different message, we can achieve this as follows:
+
+```yaml
+(...)
+---
+public_id: valory/hello_world_abci:0.1.0
+type: skill
+0:
+  models:
+    params:
+      args:
+        hello_world_message: Hello, from agent 0
+1:
+  models:
+    params:
+      args:
+        hello_world_message: Hello, from agent 1
+2:
+  models:
+    params:
+      args:
+        hello_world_message: Hello, from agent 2
+3:
+  models:
+    params:
+      args:
+        hello_world_message: Hello, from agent 3
+```
+
+!!! warning "Important"
+    If you choose to use the multiple override feature, you have to define overrides for **every** single agent that composes the service.
+
+If you have repetitive overridden parameters, you can define them using [YAML anchors](https://yaml.org/spec/1.2.2/) under the section `extra`, and reference them when defining overrides for different agents.
+
+```yaml
+(...)
+---
+extra:
+  benchmark_tool:
+    args: &id001
+      log_dir: /benchmarks
+public_id: valory/hello_world_abci:0.1.0
+type: skill
+0:
+  models:
+    params:
+      args:
+        hello_world_message: Hello, from agent 0
+    benchmark_tool:
+      args: *id001
+1:
+  models:
+    params:
+      args:
+        hello_world_message: Hello, from agent 1
+    benchmark_tool:
+      args: *id001
+2:
+  models:
+    params:
+      args:
+        hello_world_message: Hello, from agent 2
+    benchmark_tool:
+      args: *id001
+3:
+  models:
+    params:
+      args:
+        hello_world_message: Hello, from agent 3
+    benchmark_tool:
+      args: *id001
+```
+
+
+### Overrides with complicated data structures
+
+If you have nested list the environment export rules will differ as per the inner data structures. If the list is a strict list, meaning the inner values are either primitive data types like `int`, `str`, `float` or `boolean`, the list will be exported as a `json` string.
+
+
+```yaml
+(...)
+---
+extra:
+  benchmark_tool:
+    args: &id001
+      log_dir: /benchmarks
+public_id: valory/hello_world_abci:0.1.0
+type: skill
+models:
+  params:
+    args:
+      simple_list:
+        - 1
+        - 2
+        - 3
+      nested_list:
+      - - foo
+        - bar
+      - - "hello"
+        - "world"
+```
+
+These parameters will be exported as 
+
+```
+SKILL_HELLO_WORLD_ABCI_MODELS_PARAMS_ARGS_SIMPLE_LIST='[1, 2, 3]'
+SKILL_HELLO_WORLD_ABCI_MODELS_PARAMS_ARGS_NESTED_LIST='[["foo", "bar"], ["hello", "world"]]'
+```
+
+If the list contains mapping values, it'll be exported in `COMPONENT_TYPE_COMPONENT_NAME_PARAM_<MAPPING_INDEX>_KEY=value`
+
+
+```yaml
+(...)
+---
+extra:
+  benchmark_tool:
+    args: &id001
+      log_dir: /benchmarks
+public_id: valory/hello_world_abci:0.1.0
+type: skill
+models:
+  params:
+    args:
+      list_with_mappings:
+        - key: value_0
+        - key: value_1
```

Comment:
Can we have the same key twice??

Reply:
It's a list of keys, interpret it as `"list_with_mappings: [{"key": "value_0"}, {"key": "value_1"}]"`

Reply:
So this means that a dictionary {"key": "value_0"} will be passed to Agent_0, a dictionary  {"key": "value_1"} will be passed to Agent_1... and so on?

Reply:
No, they will be exported for the same agent but since they are a part of the same list they'll be exported with different index